js/src/gtag-events/utils.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * Internal dependencies
+ */
+import { sendToGroup } from './constants';
+
+/* global glaGtagData */
+
+/**
+ * Track an event using the global gtag function.
+ *
+ * @param {string} eventName
+ * @param {Object} eventParams
+ */
+export const trackEvent = ( eventName, eventParams ) => {
+	if ( typeof gtag !== 'function' ) {
+		throw new Error( 'Function gtag not implemented.' );
+	}
+	// eslint-disable-next-line no-console
+	console.log( `Tracking event ${ eventName }` );
+	window.gtag( 'event', eventName, {
+		send_to: sendToGroup,
+		...eventParams,
+	} );
+};
+
+/**
+ * Track an add_to_cart event.
+ *
+ * @param {Object} product
+ * @param {number} quantity
+ */
+export const trackAddToCartEvent = ( product, quantity = 1 ) => {
+	trackEvent( 'add_to_cart', {
+		ecomm_pagetype: 'cart',
+		event_category: 'ecommerce',
+		items: [ getItemObject( product, quantity ) ],
+	} );
+};
+
+/**
+ * Formats data into a cart Item object.
+ *
+ * @param {Object} product
+ * @param {number} quantity
+ * @return {Object} Item object.
```

Comment:
💅 Would be nice to add a `typedef` for Item Object instead of just {Object} documenting what kind of props it could have. 

https://jsdoc.app/tags-typedef.html

Reply:
Do you have some examples of how we define this for objects with optional parameters. And if the same Object is used in multiple files, then how do we share the typedef between them.

Reply:
> And if the same Object is used in multiple files, then how do we share the typedef between them.

With this syntax. (Example extracted from `js/src/components/shipping-rate-section/estimated-shipping-rates-card/rate-form-modals/rate-form-modal.js`)

```
/**
 * @typedef { import("@wordpress/components").Button } Button
 * @typedef { import(".~/data/actions").CountryCode } CountryCode
 * @typedef { import("../typedefs.js").ShippingRateGroup } ShippingRateGroup
 */
```

> Do you have some examples of how we define this for objects with optional parameters. 

With brackets. Check `sales` or `conversions` in the example below. (Example extracted from `js/src/data/utils.js`) 
You can also do `@property {fieldType} [fiedlName=value] Description` where `value` is the default value in case `fieldName` is not defined. Otherwise just `@property {fieldType} [fiedlName] Description`  should work

```
/**
 * Report fields fetched from report API.
 *
 * @typedef {Object} ReportFieldsSchema
 * @property {number} clicks Clicks value.
 * @property {number} impressions Impressions value.
 * @property {number} [sales] Sales value. Available for paid type.
 * @property {number} [conversions] Conversions value. Available for paid type.
 * @property {number} [spend] Spend value. Available for paid type.
 */
```